DIPs/DIP1xxx-Destructor.md

```diff
@@ -0,0 +1,59 @@
+# Destructor Tools
+
+| Field           | Value                                                           |
+|-----------------|-----------------------------------------------------------------|
+| DIP:            | (number/id -- assigned by DIP Manager)                          |
+| Review Count:   | 0 (edited by DIP Manager)                                       |
+| Author:         | Alexander Heistermann (alexanderheisterman@gmail.com)           |
+| Implementation: | None                                                            |
+| Status:         | Draft                                                           |
+
+## Abstract
+Opt-in static checking for destroy function and weaker destroy function named destructor_hook for classes.
+
+## Contents
+* [Rationale](#rationale)
+* [Description](#description)
+* [Reviews](#reviews)
+
+
+## Rational 
+
+The current �official� way of calling the D class deconstrutor is to call the destroy function.
+However, the destroy function cannot be called in the context of system default attributes (such as @safe or @nogc) regardless of the class attributes, thus severely restrict the usage of classes without resorting to workarounds/hack
```

Comment:
Code should be surrounded in backticks:  (e.g. \`@safe\` become `@safe`, \`__dtor\` becomes `__dtor`, etc.)